api/buffer/design.go

```diff
@@ -334,151 +337,229 @@ var tokenSecurity = APIKeySecurity("storage-api-token", func() {
 
-var ReceiverID = Type("ReceiverID", String, func() {
-	Meta("struct:field:type", "= key.ReceiverID", "github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key")
-	Description("Unique ID of the receiver.")
+var Switchable = Type("Switchable", func() {
+	Description("Switchable")
+	Attribute("disabled", Boolean, "True, if the entity is disabled.")
+	Attribute("disabledBy", String, `Who turned off the entity, for example "system", "user", ...`)
+	Attribute("disabledAt", String, func() {
+		Description("Date and time of disabling.")
+		Format(FormatDateTime)
+		Example("2022-04-28T14:20:04.000Z")
+	})
+	Attribute("disabledReason", String, "Why was the entity disabled?")
+	Required("disabled")
+})
+
+// Source -------------------------------------------------------------------------------------------------------------
+
+var SourceID = Type("SourceID", String, func() {
+	Meta("struct:field:type", "= key.SourceID", "github.com/keboola/keboola-as-code/internal/pkg/service/buffer/definition/key")
+	Description("Unique ID of the source.")
 	MinLength(1)
 	MaxLength(48)
-	Example("github-webhook-receiver")
+	Example("github-webhook-source")
+})
+
+var Source = Type("Source", func() {
+	Description(fmt.Sprintf("Source of data for further processing, start of the stream, max %d sources per a branch.", repository.MaxSourcesPerBranch))
+	Attribute("projectId", ProjectID)
+	Attribute("branchId", BranchID)
+	Attribute("sourceId", SourceID)
+	Attribute("version", EntityVersion)
+	Extend(SoftDeletable)
+	Extend(Switchable)
+	sourceFields()
+	Attribute("sinks", ArrayOf(Sink), func() {
+		Description(fmt.Sprintf("List of sinks, max %d sinks per a source.", repository.MaxSinksPerSource))
+		Example([]any{ExampleSink()})
+	})
+	Required("projectId", "branchId", "sourceId", "version", "type", "name", "description", "type", "sinks")
+	Example(ExampleSource())
 })
 
-var Receiver = Type("Receiver", func() {
-	Description("An endpoint for importing data, max 100 receivers per a project.")
-	Attribute("id", ReceiverID)
+var SourceType = Type("SourceType", String, func() {
+	Meta("struct:field:type", "= definition.SourceType", "github.com/keboola/keboola-as-code/internal/pkg/service/buffer/definition")
+	Enum(definition.SourceTypeHTTP.String())
+	Example(definition.SourceTypeHTTP.String())
+})
+
+var HTTPSource = Type("HTTPSource", func() {
+	Description("HTTP endpoint data source definition.")
 	Attribute("url", String, func() {
-		Description("URL of the receiver. Contains secret used for authentication.")
+		Description("URL of the HTTP source. Contains secret used for authentication.")
 	})
-	receiverFields()
-	Attribute("exports", ArrayOf(Export), func() {
-		Description("List of exports, max 20 exports per a receiver.")
-		Example([]any{ExampleExport()})
-	})
-	Required("id", "url", "name", "description", "exports")
-	Example(ExampleReceiver())
+	Required("url")
+	//Example(ExampleHTTPSource())
```

Comment:
if we don't need that, remove that

Reply:
:heavy_check_mark: fixed https://github.com/keboola/keboola-as-code/pull/1536/commits/49344db636079bc486bd3fc03152076137b35a2f